src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -844,6 +868,72 @@ def _get_kfp_ui_daemon_port(self) -> int:
             port = networking_utils.find_available_port()
         return port
 
+    def _get_session_cookie(self, username: str, password: str) -> str:
+        """Gets session cookie from username and password.
+
+        Args:
+            username: Username for kubeflow host.
+            password: Password for kubeflow host.
+
+        Raises:
+            ValueError: Raised when trying to fetch cookie.
+            RuntimeError: Raised when trying to fetch cookie.
+            RuntimeError: Raised when trying to fetch cookie.
+            ProvisioningError: Raised when trying to fetch cookie.
+            ProvisioningError: Raised when trying to fetch cookie.
+
+        Returns:
+            str: Cookie with the prefix `authsession=`.
+        """
+        if self.config.kubeflow_hostname is None:
+            raise ValueError(
```

Comment:
```suggestion
            raise RuntimeError(
```

So the function just raises a single error type, easier in the docstring and also to catch outside